lib/fields.c

```diff
@@ -25,41 +27,27 @@
 /*
  * valid_field - insure that a field contains all legal characters
  *
- * The supplied field is scanned for non-printable and other illegal
- * characters.
- *  + -1 is returned if an illegal or control character is present.
- *  +  1 is returned if no illegal or control characters are present,
- *       but the field contains a non-printable character.
- *  +  0 is returned otherwise.
+ * Return:
+ *	-1	Illegal or control characters are present.
+ *	1	Non-ASCII characters are present.
+ *	0	All chatacters are legal and ASCII.
  */
-int valid_field (const char *field, const char *illegal)
+int
+valid_field(const char *field, const char *illegal)
 {
-	const char *cp;
-	int err = 0;
-
-	if (NULL == field) {
+	if (NULL == field)
 		return -1;
-	}
 
-	/* For each character of field, search if it appears in the list
-	 * of illegal characters. */
-	if (illegal && strpbrk(field, illegal)) {
+	if (strpbrk(field, illegal))
```

Comment:
This seems opposite of your usual attitude  :)

Is the assumption that if null could be sent, gcc will catch it bc it's declared as a nonnull argument?

Otherwise a build_bug_on would be nice

Reply:
This string literal trick has payed off more than I thought.  I thought it would only be useful in one place, at first, but it seems widely useful.  :)